src/s3/activation.ts

```diff
@@ -30,6 +35,45 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
         }),
         vscode.commands.registerCommand('aws.s3.uploadFile', async (node: S3BucketNode | S3FolderNode) => {
             await uploadFileCommand(node)
+        }),
+        vscode.commands.registerCommand('aws.s3.uploadFileToS3', async () => {
+            while (true) {
+                const editor = vscode.window.activeTextEditor
+                const document = editor?.document.uri
+                const file = await getFileToUpload(document)
+                if (file) {
+                    let bucketResponse
+                    const regionCode = Workspace.vscode().getConfiguration(extensionSettingsPrefix).get<string>('s3.defaultRegion')?? "us-east-1"
+                    const s3Client = ext.toolkitClientBuilder.createS3Client(regionCode)
+                    try{
+                        bucketResponse = await promptUserForBucket(s3Client)
+                    } catch (e) {
+                        telemetry.recordS3UploadObject({ result: 'Failed' })
+                        return
+                    }
+                    if (bucketResponse === 'back'){
+                        continue
+                    }
+                    if(bucketResponse == 'cancel'){
+                        getLogger().info('No bucket selected, cancelling upload')
+                        telemetry.recordS3UploadObject({ result: 'Failed' })
+                        return
+                    }
```

Comment:
Wrap all of this in the try statement, that way you can declare the bucketResponse as a const (since the scope won't have to leave the try statement)

Reply:
I would not say that  `const` is always  worth the loss of `try` granularity